File: Massive project.ts
/**
 * Advanced Hello World — TypeScript Edition
 * ==========================================
 * Multiple ways to say "Hello, World!"
 * Run: npx ts-node hello_world_advanced.ts [mode] [name]
 * Or compile: tsc hello_world_advanced.ts && node hello_world_advanced.js
 */

// ─────────────────────────────────────────────
// Types
// ─────────────────────────────────────────────
type Language = keyof typeof TRANSLATIONS;

interface GreetingResult {
  language: string;
  message: string;
  timestamp: string;
}

// ─────────────────────────────────────────────
// 1. Classic
// ─────────────────────────────────────────────
function helloClassic(): void {
  console.log("Hello, World!");
}

// ─────────────────────────────────────────────
// 2. Personalised
// ─────────────────────────────────────────────
function helloPersonal(name: string = "World"): void {
  console.log(`Hello, ${name}!`);
}

// ─────────────────────────────────────────────
// 3. Time-aware
// ─────────────────────────────────────────────
function helloTime(): void {
  const hour = new Date().getHours();
  const period =
    hour < 12 ? "Morning" : hour < 18 ? "Afternoon" : "Evening";
  const time = new Date().toLocaleTimeString();
  console.log(`Good ${period}, World! 🌍  [${time}]`);
}

// ─────────────────────────────────────────────
// 4. Multilingual
// ─────────────────────────────────────────────
const TRANSLATIONS = {
  English:    "Hello, World!",
  Spanish:    "¡Hola, Mundo!",
  French:     "Bonjour, le Monde !",
  German:     "Hallo, Welt!",
  Japanese:   "こんにちは、世界！",
  Arabic:     "مرحبا بالعالم!",
  Swahili:    "Habari, Dunia!",
  Portuguese: "Olá, Mundo!",
} as const;

function helloMultilingual(): void {
  for (const [lang, greeting] of Object.entries(TRANSLATIONS)) {
    console.log(`  ${lang.padEnd(12)} → ${greeting}`);
  }
}

// ─────────────────────────────────────────────
// 5. Random language
// ─────────────────────────────────────────────
function helloRandom(): void {
  const entries = Object.entries(TRANSLATIONS);
  const [lang, greeting] = entries[Math.floor(Math.random() * entries.length)];
  console.log(`[${lang}] ${greeting}`);
}

// ─────────────────────────────────────────────
// 6. Typewriter effect
// ─────────────────────────────────────────────
async function helloTypewriter(
  message: string = "Hello, World!",
  delay: number = 70
): Promise<void> {
  for (const char of message) {
    process.stdout.write(char);
    await new Promise((resolve) => setTimeout(resolve, delay));
  }
  console.log();
}

// ─────────────────────────────────────────────
// 7. ASCII art banner
// ─────────────────────────────────────────────
function helloBanner(): void {
  const banner = `
 _   _      _ _        __        __         _     _   _
| | | | ___| | | ___   \\ \\      / /__  _ __| | __| | | |
| |_| |/ _ \\ | |/ _ \\   \\ \\ /\\ / / _ \\| '__| |/ _\` | | |
|  _  |  __/ | | (_) |   \\ V  V / (_) | |  | | (_| | |_|
|_| |_|\\___|_|_|\\___/     \\_/\\_/ \\___/|_|  |_|\\__,_| (_)
`;
  console.log(banner);
}

// ─────────────────────────────────────────────
// 8. Countdown then greet
// ─────────────────────────────────────────────
async function helloCountdown(start: number = 3): Promise<void> {
  for (let i = start; i > 0; i--) {
    console.log(`  ${i}…`);
    await new Promise((resolve) => setTimeout(resolve, 600));
  }
  console.log("Hello, World! 🎉");
}

// ─────────────────────────────────────────────
// 9. Return instead of print (useful for APIs / tests)
// ─────────────────────────────────────────────
function helloReturn(name: string = "World"): string {
  return `Hello, ${name}!`;
}

// ─────────────────────────────────────────────
// 10. Structured object (useful for APIs)
// ─────────────────────────────────────────────
function helloObject(language: Language = "English"): GreetingResult {
  return {
    language,
    message: TRANSLATIONS[language],
    timestamp: new Date().toISOString(),
  };
}

// ─────────────────────────────────────────────
// CLI
// ─────────────────────────────────────────────
const MODES: Record<string, () => void | Promise<void>> = {
  classic:      helloClassic,
  time:         helloTime,
  multilingual: helloMultilingual,
  random:       helloRandom,
  typewriter:   helloTypewriter,
  banner:       helloBanner,
  countdown:    helloCountdown,
  object:       () => console.log(JSON.stringify(helloObject(), null, 2)),
};

async function main(): Promise<void> {
  const args = process.argv.slice(2);

  if (args.length === 0) {
    // Run all modes
    for (const [name, fn] of Object.entries(MODES)) {
      console.log(`\n── ${name.toUpperCase()} ──`);
      await fn();
    }
    console.log("\n── PERSONAL ──");
    helloPersonal("GitHub");

    console.log("\n── RETURN (no print) ──");
    const msg = helloReturn("TypeScript");
    console.log(`Returned: "${msg}"`);
    return;
  }

  const mode = args[0].toLowerCase();

  if (mode === "personal") {
    helloPersonal(args[1] ?? "World");
  } else if (mode in MODES) {
    await MODES[mode]();
  } else {
    console.error(
      `Unknown mode '${mode}'. Available: ${Object.keys(MODES).join(", ")}, personal`
    );
    process.exit(1);
  }
}

main();

// ─────────────────────────────────────────────
// Exports (for use as a module)
// ─────────────────────────────────────────────
export {
  helloClassic,
  helloPersonal,
  helloTime,
  helloMultilingual,
  helloRandom,
  helloTypewriter,
  helloBanner,
  helloCountdown,
  helloReturn,
  helloObject,
  TRANSLATIONS,
};
export type { GreetingResult, Language };

File: Massive project.py
"""
Advanced Hello World
====================
Multiple ways to say "Hello, World!" in Python.
Run directly or import individual functions.
"""

import sys
import time
import random
import textwrap
from datetime import datetime


# ─────────────────────────────────────────────
# 1. Classic
# ─────────────────────────────────────────────
def hello_classic():
    """The original."""
    print("Hello, World!")


# ─────────────────────────────────────────────
# 2. Personalised
# ─────────────────────────────────────────────
def hello_personal(name: str = "World"):
    """Greet a specific person (or the whole world by default)."""
    print(f"Hello, {name}!")


# ─────────────────────────────────────────────
# 3. Time-aware
# ─────────────────────────────────────────────
def hello_time():
    """Good morning / afternoon / evening greeting."""
    hour = datetime.now().hour
    if hour < 12:
        period = "Morning"
    elif hour < 18:
        period = "Afternoon"
    else:
        period = "Evening"
    print(f"Good {period}, World! 🌍  [{datetime.now().strftime('%H:%M:%S')}]")


# ─────────────────────────────────────────────
# 4. Multilingual
# ─────────────────────────────────────────────
TRANSLATIONS = {
    "English":    "Hello, World!",
    "Spanish":    "¡Hola, Mundo!",
    "French":     "Bonjour, le Monde !",
    "German":     "Hallo, Welt!",
    "Japanese":   "こんにちは、世界！",
    "Arabic":     "مرحبا بالعالم!",
    "Swahili":    "Habari, Dunia!",
    "Portuguese": "Olá, Mundo!",
}

def hello_multilingual():
    """Print Hello World in every supported language."""
    for lang, greeting in TRANSLATIONS.items():
        print(f"  {lang:<12} → {greeting}")


# ─────────────────────────────────────────────
# 5. Random language
# ─────────────────────────────────────────────
def hello_random():
    """Pick a surprise language."""
    lang, greeting = random.choice(list(TRANSLATIONS.items()))
    print(f"[{lang}] {greeting}")


# ─────────────────────────────────────────────
# 6. Typewriter effect
# ─────────────────────────────────────────────
def hello_typewriter(message: str = "Hello, World!", delay: float = 0.07):
    """Print one character at a time."""
    for char in message:
        print(char, end="", flush=True)
        time.sleep(delay)
    print()


# ─────────────────────────────────────────────
# 7. ASCII art banner
# ─────────────────────────────────────────────
def hello_banner():
    """Big block-letter banner."""
    banner = textwrap.dedent(r"""
     _   _      _ _        __        __         _     _   _
    | | | | ___| | | ___   \ \      / /__  _ __| | __| | | |
    | |_| |/ _ \ | |/ _ \   \ \ /\ / / _ \| '__| |/ _` | | |
    |  _  |  __/ | | (_) |   \ V  V / (_) | |  | | (_| | |_|
    |_| |_|\___|_|_|\___/     \_/\_/ \___/|_|  |_|\__,_| (_)
    """)
    print(banner)


# ─────────────────────────────────────────────
# 8. Countdown then greet
# ─────────────────────────────────────────────
def hello_countdown(start: int = 3):
    """Countdown, then say hello."""
    for i in range(start, 0, -1):
        print(f"  {i}…", flush=True)
        time.sleep(0.6)
    print("Hello, World! 🎉")


# ─────────────────────────────────────────────
# 9. Return instead of print (useful for APIs / tests)
# ─────────────────────────────────────────────
def hello_return(name: str = "World") -> str:
    """Return the greeting as a string instead of printing it."""
    return f"Hello, {name}!"


# ─────────────────────────────────────────────
# 10. Command-line interface
# ─────────────────────────────────────────────
MODES = {
    "classic":      hello_classic,
    "time":         hello_time,
    "multilingual": hello_multilingual,
    "random":       hello_random,
    "typewriter":   hello_typewriter,
    "banner":       hello_banner,
    "countdown":    hello_countdown,
}

def main():
    """
    Usage:
        python hello_world_advanced.py                  # runs all modes
        python hello_world_advanced.py classic          # one mode
        python hello_world_advanced.py personal Alice   # personalised
    """
    args = sys.argv[1:]

    if not args:
        # Run every mode with a header
        for name, fn in MODES.items():
            print(f"\n── {name.upper()} ──")
            fn()
        print(f"\n── PERSONAL ──")
        hello_personal("GitHub")
        return

    mode = args[0].lower()

    if mode == "personal":
        name = args[1] if len(args) > 1 else "World"
        hello_personal(name)
    elif mode in MODES:
        MODES[mode]()
    else:
        print(f"Unknown mode '{mode}'. Available: {', '.join(MODES)} personal")
        sys.exit(1)


if __name__ == "__main__":
    main()
